content/en/blog/2024/spring-starter-stable/index.md

```diff
@@ -0,0 +1,690 @@
+[lists](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/release/v2.6.x/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java#L104-L106)
+and
+[maps](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/release/v2.6.x/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java#L126-L140),
+because the Spring Boot Environment can't handle them directly. Luckily, Spring
+Boot has a way to convert Strings to lists and
+[maps](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/release/v2.6.x/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/MapConverter.java),
+so you can pass resource attributes both in a single environment variable (as
+per [spec](/docs/languages/sdk-configuration/general/#otel_resource_attributes))
+or in a
+[Spring Boot configuration file](/docs/zero-code/java/spring-boot-starter/sdk-configuration/#general-configuration).
+
+### Programmatic SDK auto-configuration setup
+
+Spring Boot users also expect to be able to use Spring beans for advanced configuration.
+The SDK auto-configuration did not know about Spring beans, so we came up with a new interface,
+[ComponentLoader](https://github.com/open-telemetry/opentelemetry-java/blob/release/v1.40.x/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/internal/ComponentLoader.java),
+that allows you to register Spring beans that will be loaded by the
+OpenTelemetry SDK auto-configuration.
+
+The [Spring Starter implementation of the ComponentLoader interface](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java#L162-L181) 
+uses Spring's `ApplicationContext` to find all beans of a certain type.
+
+This allows you to register your own customizers, and other SDK components as Spring beans.
+You can find the available SPI interfaces implementable as Spring beans in the
+[OpenTelemetry SDK auto-configuration SPI documentation](/docs/languages/java/configuration/#spi-service-provider-interface).
+
+You should be able to customize most aspects of the OpenTelemetry SDK by implementing a bean that returns an 
+[AutoConfigurationCustomizerProvider](/docs/languages/java/configuration/#autoconfigurationcustomizerprovider) 
+instance - an idiomatic Spring Boot approach to customization.
+
+This example shows how to create a bean that customizes the sampler to drop spans for paths starting with `/actuator`:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/FilterPaths.java"?>
+```java
+package otel;
```

Comment:
I don't think we should use code excerpts with blog posts: they are a snapshot in time, and we accept them to outdated eventually. Otherwise we create a lot of maintenance overhead to keep them updated.